spec/solver_spec.rb

```diff
@@ -0,0 +1,43 @@
+require_relative '../solver'
+
+describe Solver do
+  before(:each) do
+    @solver = Solver.new
+  end
+
+  context 'factorial method' do
+    it 'take N and return factorial for that number' do
+      result = @solver.factorial(0)
+      expect(result).to eq 1
+    end
+  end
```

Comment:
- [ ] For the factorial, please write a test that can handle a negative value which raises an exception as an expectation 👍

Reply:
Already implemented it! Thanks for the feedback! :smile: